### PR TITLE
Refact/keyp 347 rename core classes

### PR DESCRIPTION
**Pull request by jeanpierrefortune**
Mainly a renaming step

### CODE REVIEW

File: java/component/keyple-core/src/main/java/org/eclipse/keyple/core/selection/SelectionsResult.java [hunk 1]
```diff
@@ -17,64 +17,64 @@
...
     }
-    return matchingSe;
+    return smartCard;
   }
 
-  /** @return the {@link AbstractMatchingSe} map */
-  public Map<Integer, AbstractMatchingSe> getMatchingSelections() {
-    return matchingSeMap;
+  /** @return the {@link AbstractSmartCard} map */
+  public Map<Integer, AbstractSmartCard> getMatchingSmartCards() {
```
Comment (andrei-cristea) [inline java/component/keyple-core/src/main/java/org/eclipse/keyple/core/selection/SelectionsResult.java:64]: retirer matching

File: java/component/keyple-core/src/main/java/org/eclipse/keyple/core/seproxy/Reader.java [hunk 2]
```diff
@@ -19,21 +19,21 @@
...
-   * Checks if is SE present.
+   * Checks if is the card present.
    *
-   * @return true if a Secure Element is present in the reader
-   * @throws KeypleReaderIOException if the communication with the reader or the SE has failed
+   * @return true if a card is present in the reader
+   * @throws KeypleReaderIOException if the communication with the reader or the card has failed
    * @since 0.9
    */
   boolean isSePresent();
```
Comment (andrei-cristea) [inline java/component/keyple-core/src/main/java/org/eclipse/keyple/core/seproxy/Reader.java:39]: moche

File: java/component/keyple-core/src/test/java/org/eclipse/keyple/core/seproxy/message/CardRequestTest.java [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+
+  @Test
+  public void getApduRequests() {
+    // test
+    cardRequest = new CardRequest(getSelector(null), apdus);
+    assertArrayEquals(apdus.toArray(), cardRequest.getApduRequests().toArray());
+  }
+
+  @Test
+  public void getSeProtocol() {
```
Comment (andrei-cristea) [inline java/component/keyple-core/src/test/java/org/eclipse/keyple/core/seproxy/message/CardRequestTest.java:73]: honte

File: java/component/keyple-core/src/test/java/org/eclipse/keyple/core/seproxy/plugin/reader/AbsLocalReaderTransmitTest.java [hunk 4]
```diff
@@ -210,17 +211,17 @@ public void transmit_partial_response_3() throws Exception {
...
+    CardSelector selector =
+        CardSelector.builder()
             .seProtocol(ContactlessCardCommonProtocols.ISO_14443_4.name())
             .atrFilter(atrFilter)
             .build();
 
-    SeSelector failSelector =
-        SeSelector.builder().seProtocol("MIFARE_ULTRA_LIGHT").atrFilter(atrFilter).build();
+    CardSelector failSelector =
+        CardSelector.builder().seProtocol("MIFARE_ULTRA_LIGHT").atrFilter(atrFilter).build();
```
Comment (andrei-cristea) [inline java/component/keyple-core/src/test/java/org/eclipse/keyple/core/seproxy/plugin/reader/AbsLocalReaderTransmitTest.java:224]: seProtocol

File: java/component/keyple-core/src/test/java/org/eclipse/keyple/core/seproxy/plugin/reader/WaitForSeInsertionStateJobExecutorTest.java [hunk 5]
```diff
@@ -111,9 +111,9 @@ public void testInsertSe_NotMatched() throws Exception {
   @Test
   public void testInsertSe_Error() throws Exception {
     /*
-     * input SE inserted SE doesnt matched
+     * input card inserted card doesnt matched
      */
-    // se not matched
+    // card not matched
     doReturn(null).when(r).processSeInserted();
```
Comment (andrei-cristea) [inline java/component/keyple-core/src/test/java/org/eclipse/keyple/core/seproxy/plugin/reader/WaitForSeInsertionStateJobExecutorTest.java:117]: oulala

File: java/component/keyple-core/src/test/java/org/eclipse/keyple/core/seproxy/plugin/reader/WaitForSeRemovalStateTest.java [hunk 6]
```diff
@@ -49,7 +49,7 @@ public void setUp() {
   @Test
   public void waitForRemoval_SINGLESHOT() throws Exception {
     /*
-     * ------------ input polling mode is STOP SE has been removed within timeout
+     * ------------ input polling mode is STOP card has been removed within timeout
      */
     AbstractObservableLocalReader r = AbsSmartInsertionTheadedReaderTest.getMock(READER_NAME);
     WaitForSeRemovalState waitForSeRemovalState = new WaitForSeRemovalState(r);
```
Comment (andrei-cristea) [inline java/component/keyple-core/src/test/java/org/eclipse/keyple/core/seproxy/plugin/reader/WaitForSeRemovalStateTest.java:55]: oups

File: java/example/generic/pc/Demo_SeProtocolDetection/src/main/java/org/eclipse/keyple/example/generic/pc/Demo_SeProtocolDetection/SeProtocolDetectionEngine.java [hunk 7]
```diff
@@ -82,29 +82,29 @@ public AbstractDefaultSelectionsRequest prepareSeSelection() {
           break;
         default:
           /* Add a generic selector */
-          seSelection.prepareSelection(
-              new GenericSeSelectionRequest(
-                  SeSelector.builder()
+          cardSelection.prepareSelection(
+              new GenericCardSelectionRequest(
+                  CardSelector.builder()
                       .seProtocol(ContactlessCardCommonProtocols.ISO_14443_4.name())
```
Comment (andrei-cristea) [inline java/example/generic/pc/Demo_SeProtocolDetection/src/main/java/org/eclipse/keyple/example/generic/pc/Demo_SeProtocolDetection/SeProtocolDetectionEngine.java:88]: oulala

File: java/example/generic/pc/UseCase3_GroupedMultiSelection/src/main/java/org/eclipse/keyple/example/generic/pc/usecase3/GroupedMultiSelection_Pcsc.java [hunk 8]
```diff
@@ -37,90 +37,91 @@
...
-    if (seReader.isSePresent()) {
+    // Check if a card is present in the reader
+    if (reader.isSePresent()) {
 
-      SeSelection seSelection = new SeSelection(MultiSeRequestProcessing.PROCESS_ALL);
+      CardSelection cardSelection = new CardSelection(MultiSelectionProcessing.PROCESS_ALL);
 
-      // operate SE selection (change the AID here to adapt it to the SE used for the test)
+      // operate the card selection (change the AID here to adapt it to the card used for the test)
       String seAidPrefix = "A000000404012509";
```
Comment (andrei-cristea) [inline java/example/generic/pc/UseCase3_GroupedMultiSelection/src/main/java/org/eclipse/keyple/example/generic/pc/usecase3/GroupedMultiSelection_Pcsc.java:61]: ouh c'est moche

**Review by andrei-cristea — CHANGES_REQUESTED** [review]

File: java/component/keyple-core/src/main/java/org/eclipse/keyple/core/selection/SelectionsResult.java (hunk 1 above)
Comment (jeanpierrefortune) [reply java/component/keyple-core/src/main/java/org/eclipse/keyple/core/selection/SelectionsResult.java:64]: getMatchingSmartCards -> getSmartCards

File: java/component/keyple-core/src/main/java/org/eclipse/keyple/core/seproxy/Reader.java (hunk 2 above)
Comment (jeanpierrefortune) [reply java/component/keyple-core/src/main/java/org/eclipse/keyple/core/seproxy/Reader.java:39]: isSePresent -> isCardPresent

File: java/component/keyple-core/src/test/java/org/eclipse/keyple/core/seproxy/message/CardRequestTest.java (hunk 3 above)
Comment (jeanpierrefortune) [reply java/component/keyple-core/src/test/java/org/eclipse/keyple/core/seproxy/message/CardRequestTest.java:73]: getSeProtocol -> getCardProtocol

File: java/component/keyple-core/src/test/java/org/eclipse/keyple/core/seproxy/plugin/reader/AbsLocalReaderTransmitTest.java (hunk 4 above)
Comment (jeanpierrefortune) [reply java/component/keyple-core/src/test/java/org/eclipse/keyple/core/seproxy/plugin/reader/AbsLocalReaderTransmitTest.java:224]: seProtocol -> cardProtocol

File: java/component/keyple-core/src/test/java/org/eclipse/keyple/core/seproxy/plugin/reader/WaitForSeInsertionStateJobExecutorTest.java (hunk 5 above)
Comment (jeanpierrefortune) [reply java/component/keyple-core/src/test/java/org/eclipse/keyple/core/seproxy/plugin/reader/WaitForSeInsertionStateJobExecutorTest.java:117]: processSeInserted -> processCardInserted

File: java/component/keyple-core/src/test/java/org/eclipse/keyple/core/seproxy/plugin/reader/WaitForSeRemovalStateTest.java (hunk 6 above)
Comment (jeanpierrefortune) [reply java/component/keyple-core/src/test/java/org/eclipse/keyple/core/seproxy/plugin/reader/WaitForSeRemovalStateTest.java:55]: All names like WaitForSeXyz changed to WaitForCardXyz

File: java/example/generic/pc/UseCase3_GroupedMultiSelection/src/main/java/org/eclipse/keyple/example/generic/pc/usecase3/GroupedMultiSelection_Pcsc.java (hunk 8 above)
Comment (jeanpierrefortune) [reply java/example/generic/pc/UseCase3_GroupedMultiSelection/src/main/java/org/eclipse/keyple/example/generic/pc/usecase3/GroupedMultiSelection_Pcsc.java:61]: all names like seXyz renamed cardXyz

File: java/example/generic/pc/Demo_SeProtocolDetection/src/main/java/org/eclipse/keyple/example/generic/pc/Demo_SeProtocolDetection/SeProtocolDetectionEngine.java (hunk 7 above)
Comment (jeanpierrefortune) [reply java/example/generic/pc/Demo_SeProtocolDetection/src/main/java/org/eclipse/keyple/example/generic/pc/Demo_SeProtocolDetection/SeProtocolDetectionEngine.java:88]: seProtocol -> cardProtocol

File: android/keyple-plugin/android-omapi/src/main/kotlin/org/eclipse/keyple/plugin/android/omapi/se/AndroidOmapiPlugin.kt [hunk 9]
```diff
@@ -25,19 +23,19 @@ import timber.log.Timber
...
         Timber.i("OMAPI SEService version: %s", seService?.version)
     }
 
-    override fun getNativeReaders(): Array<Reader>? {
+    override fun getNativeReaders(): Array<android.se.omapi.Reader>? {
         return seService?.readers
     }
 
-    override fun mapToSeReader(nativeReader: Reader): SeReader {
+    override fun mapToSeReader(nativeReader: android.se.omapi.Reader): org.eclipse.keyple.core.seproxy.Reader {
```
[the remark this answers is not in the document]
Comment (jeanpierrefortune) [reply android/keyple-plugin/android-omapi/src/main/kotlin/org/eclipse/keyple/plugin/android/omapi/se/AndroidOmapiPlugin.kt]: changed imports

File: android/keyple-plugin/android-omapi/src/main/kotlin/org/eclipse/keyple/plugin/android/omapi/simalliance/AndroidOmapiPlugin.kt [hunk 10]
```diff
@@ -12,30 +12,28 @@
...
         Timber.i("OMAPI SEService version: %s", seService?.version)
     }
 
-    override fun getNativeReaders(): Array<Reader>? {
+    override fun getNativeReaders(): Array<org.simalliance.openmobileapi.Reader>? {
         return seService?.readers
     }
 
-    override fun mapToSeReader(nativeReader: Reader): SeReader {
+    override fun mapToSeReader(nativeReader: org.simalliance.openmobileapi.Reader): org.eclipse.keyple.core.seproxy.Reader {
```
[the remark this answers is not in the document]
Comment (jeanpierrefortune) [reply android/keyple-plugin/android-omapi/src/main/kotlin/org/eclipse/keyple/plugin/android/omapi/simalliance/AndroidOmapiPlugin.kt]: changed imports

File: ReleaseNotes.md [hunk 11]
```diff
@@ -217,7 +217,7 @@ Release train components:
 * SE Proxy API: generic universal interface to interface a smartcard reader; transmission of grouped APDU commands; observable reader notification
 * Transaction API: management of an explicit application selection on a reader; setting of a default selection operation on an observed reader
 * Known issues/limits 
-  * Allows the cast of ReaderPlugin & SeReader native classes: could cause a misusage of the API by ticketing application. A ReaderPlugin factory would be necessary for the setting of plugins to the SE Proxy Service.
+  * Allows the cast of Plugin & SeReader native classes: could cause a misusage of the API by ticketing application. A Plugin factory would be necessary for the setting of plugins to the SE Proxy Service.
```
Comment (pterr) [inline ReleaseNotes.md]: Allows the cast of Plugin & **Reader**

**Review by pterr — APPROVED** [review]

**Pull request merged**